### PR TITLE
fix: Support inserting models with only id field.

### PR DESCRIPTION
**Pull request by SandPod**
### Changes:
- Moves the logic around building insert statement into the SQL query builder for improved testability.
- Use `'INSERT INTO "${_table.tableName}" DEFAULT VALUES RETURNING *` as suggested by [PostgreSQL documentation](https://www.postgresql.org/docs/16/dml-insert.html) for inserts with only default values.

Closes: #2306 

### Review suggestion:
Review commit by commit, since there was a refactoring and some model generation this should help greatly.

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [x] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

None - Fixes a bug.

### CODE REVIEW

File: tests/serverpod_test_server/lib/src/models/empty_model/empty_model_relation_item.spy.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class: EmptyModelRelationItem
+table: empty_model_relation_item
+fields:
+  name: String
+  containerId: int, relation(parent=empty_model)
```
Comment (Isakdl) [inline tests/serverpod_test_server/lib/src/models/empty_model/empty_model_relation_item.spy.yaml:5]: Comment: Note that we have two relations in these two models.

**Review by Isakdl — APPROVED** [review]
LGTM 🚀

File: tests/serverpod_test_server/lib/src/models/empty_model/empty_model_relation_item.spy.yaml (hunk 1 above)
Comment (SandPod) [reply tests/serverpod_test_server/lib/src/models/empty_model/empty_model_relation_item.spy.yaml:5]: Good note! Will fix that 👍

Comment (SandPod) [reply tests/serverpod_test_server/lib/src/models/empty_model/empty_model_relation_item.spy.yaml]: Done!

**Pull request merged**